src/framework/execution/dependencyGraph.ts

```diff
@@ -225,18 +241,30 @@ function buildStepContext(
   };
 }
 
-function buildStepResultsMap(dependencyGraph: DepGraph<IntegrationStep>) {
+function buildStepResultsMap(
+  dependencyGraph: DepGraph<IntegrationStep>,
+  stepStartStates: IntegrationStepStartStates,
+) {
   const stepResultMapEntries = dependencyGraph
     .overallOrder()
     .map(
       (stepId): IntegrationStepResult => {
         const step = dependencyGraph.getNodeData(stepId);
+
+        const hasDisabledDependencies =
+          dependencyGraph
+            .dependenciesOf(step.id)
+            .filter((id) => stepStartStates[id].disabled).length > 0;
+
         return {
           id: step.id,
           name: step.name,
           types: step.types,
           dependsOn: step.dependsOn,
-          status: IntegrationStepResultStatus.DISABLED,
+          status:
+            stepStartStates[step.id].disabled || hasDisabledDependencies
```

Comment:
Ah, okay that might make my `--step` change not work as I thought it would.  I was picturing if a step was passed into `--step` it would run that step and all dependent steps but I think I'll need to resolve the deps and add that to the enabled.